Sources/AuthFoundation/JWT/Internal/DefaultIDTokenValidator.swift

```diff
@@ -16,59 +16,212 @@ import Foundation
 import CommonCrypto
 #endif
 
+/// This represents the possible option using in validating a claim
+public struct ValidationOption: OptionSet  {
```

Comment:
This seems a bit verbose to me. It feels like a lot more code than the original implementation.

What's wrong with using CaseIterable?

```swift
struct DefaultIDTokenValidator: IDTokenValidator {
    var checks: [ValidationCheck] = ValidationCheck.allCases
    
    enum ValidationCheck: CaseIterable {
        case issuer, audience, scheme, algorithm, expirationTime, issuedAtTime, nonce, authTime, maxAge, subject
    }

   ...

       func validate(token: JWT, issuer: URL, clientId: String, context: IDTokenValidatorContext?) throws {
        for check in checks {
          // Put existing checks here
        }
```